app/services/authorizer.rb

```diff
@@ -12,8 +12,9 @@ def can?(permission, subject = nil)
       user.permissions.where(:name => permission).present?
     else
       return true if user.admin?
-      collection = @cache[subject.class.to_s][permission] ||= find_collection(subject.class, :permission => permission)
-      collection.include?(subject)
+      @cache["#{subject.class.to_s}-#{permission}-#{subject.id}"] ||=
+        find_collection(subject.class, :permission => permission).
+        where(:id => subject.id)
```

Comment:
I think this new `where(:id..`) behaviour should be optional, keeping the original caching of the whole resource list for the regular case of one permission, multiple resources - e.g. permission checking a button on an index page.  A per-resource cache will be ineffective when the resource ID changes, so it'll generate 2*N SQL queries or similar on those pages.

Perhaps add an options hash to the `can?` method to specify the optimal caching strategy?